lab-khalid/model/user.js

```diff
@@ -0,0 +1,75 @@
+'use strict';
+
+const
+  mongoose = require('mongoose'),
+  debug = require('debug')('CFgram:user'),
+  Promise = require('bluebird'),
+  bcrypt = require('bcrypt'),
+  crypto = require('crypto'),
+  createError = require('http-errors'),
+  jwt = require('jsonwebtoken');
+
+
+
+const userSchema = mongoose.Schema({
+  username : {type: String, required: true, unique: true},
+  email : {type: String, required: true, unique: true},
+  password: {type: String, requires: true},
+  findHash: {type: String, required: true}
```

Comment:
the findHash needs to be `unique` not `required`. Setting this to be required will cause and error when you try to save the user since they won't have a `findHash` value yet since you don't assign that till after.